compiler/rustc_mir_transform/src/gvn.rs

```diff
@@ -876,14 +876,38 @@ impl<'body, 'tcx> VnState<'body, 'tcx> {
         None
     }
 
+    fn try_as_place_elem(
+        &mut self,
+        proj: ProjectionElem<VnIndex, Ty<'tcx>>,
+        loc: Location,
+    ) -> Option<PlaceElem<'tcx>> {
+        Some(match proj {
+            ProjectionElem::Deref => ProjectionElem::Deref,
+            ProjectionElem::Field(idx, ty) => ProjectionElem::Field(idx, ty),
+            ProjectionElem::Index(idx) => {
+                return self.try_as_local(idx, loc).map(|l| ProjectionElem::Index(l));
```

Comment:
You need to insert `l` into `reused_locals`. Could you add a test case with this projection?

Reply:
Oops, thanks!